### DISCUSSION TITLE
Writing json file using Python

### DISCUSSION THREAD

**Discussion by skulimerp** [helloSystem/hello#572] [General]
please help just learned python and I don't know what I'm doing wrong I'm trying to make a new json file
my code is below I'm using VS and I don't know if its VS or me but when I run it nothing happens sometimes it will tell me "'list' object has no attribute 'json'" i know its dumb but i need help understanding if anyone has any tips please let me know 

import json

stuff = ["items", "other things"]

with open(stuff.json, "w") as outfile:
    json.dump(stuff, outfile)

**Comment by probonopd** [comment]
Hello. This is not a Python support forum, but here is what ChatGPT says:

Your code has a couple of issues. Firstly, you are trying to use `stuff.json` as if it is an attribute of the `stuff` list, which will cause an error since `stuff` is a list and lists don't have attributes like `json`. You should use a string for the file name. Secondly, the indentation for the `json.dump` line is incorrect. Here's the corrected version of your code:

```python
import json

stuff = ["items", "other things"]

# Use a string for the filename
filename = "stuff.json"

# Correct the indentation for the json.dump
with open(filename, "w") as outfile:
    json.dump(stuff, outfile)
```

This code will create a file named `stuff.json` in the current directory and write the `stuff` list to it in JSON format.

To run this script in VS Code:
1. Save the file with a `.py` extension (e.g., `create_json.py`).
2. Open a terminal in VS Code.
3. Navigate to the directory where your script is saved.
4. Run the script by typing `python create_json.py` and pressing Enter.

You should now see a `stuff.json` file in the same directory as your script. This file will contain the JSON representation of your `stuff` list:

```json
["items", "other things"]
```

**Reply by skulimerp** [reply]
thank you so much your  a life saver

**Comment by VyvaHart** [comment]
Hi @skulimerp 

You all doing good! but you need to separate the **stuff.json** file name with double quotes(""), because in this quotes you provide path to file you want to read ("r") or modify ("w")

here is the correct version:
```
import json

stuff = ["items", "other things"]

with open("stuff.json", "w") as outfile:
    json.dump(stuff, outfile)
```
after script execution you will notice that a **stuff.json** file appeared in root path

Good Luck!

**Reply by skulimerp** [reply]
thank you ive been struggling with this and it felt so silly but maybe thats just programming, getting mad over a syntax

**Reply by VyvaHart** [reply]
That's okay, the main thing is don’t worry about this bro, such things will be intuitively clear to you in the future ;)
